var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -46,6 +46,13 @@ class Petsc(Package):
     version('3.5.1', 'a557e029711ebf425544e117ffa44d8f')
     version('3.4.4', '7edbc68aa6d8d6a3295dd5f6c2f6979d')
 
+    #instead of needing to manually provide specific git tags in the package.py for each package could there 
+    #   be a way to describe the git tag/hash in the spack spec such a
+    #   @git:tag_or_commit_hash using maybe
+    #version('*', git='https://bitbucket.org/petsc/petsc.git')
+    #   then users could do petsc@master or petsc@v3.7.1 or petsc@11bc23.. ?
+    version('master', git='https://bitbucket.org/petsc/petsc.git')
```

Comment:
if you use `master`, in comparators for versions the following won't work: `master > 3.7.2` and alike. For now you should name it `develop` to behave correctly. I have a PR https://github.com/LLNL/spack/pull/1983 which introduce other aliases for infinity-like versions, i.e. those which are greater than any given release, but there is a lot of discussion around Version and different opinions.

ps. The proper comparision is important if packages write `depends_on(petsc@3.7.0:` (open range) or `depends_on(petsc@:3.6.4)` (again open range).